pre_commit/xargs.py

```diff
@@ -43,12 +52,23 @@ def partition(cmd, varargs, _max_length=MAX_LENGTH):
     return tuple(ret)
 
 
+@contextlib.contextmanager
+def _threadpool(size):
```

Comment:
@asottile let me know if you have a preference on doing this vs installing a concurrent.futures backport on python2

Reply:
up to you, `concurrent.futures` is slightly more future-proof if the code is simpler (if it isn't then definitely don't bother)